.github/workflows/int-and-connected-test-run.yml

```diff
@@ -145,44 +194,71 @@ jobs:
           echo "::set-output name=git_hash::$git_hash"
           echo "Latest git hash for this branch: $git_hash"
       - name: "Wait for deployment to come back online"
-        uses: broadinstitute/datarepo-actions/actions/wait-for-deployment@0.62.0
+        uses: broadinstitute/datarepo-actions/actions/wait-for-deployment@dr-2757-nm-better-unit
         timeout-minutes: 20
         env:
           DESIRED_GITHASH: ${{ steps.configuration.outputs.git_hash }}
           DEPLOYMENT_TYPE: 'api'
       - name: "Run Test Runner smoke tests via Gradle"
-        uses: broadinstitute/datarepo-actions/actions/main@0.62.0
+        uses: broadinstitute/datarepo-actions/actions/main@dr-2757-nm-better-unit
         with:
           actions_subcommand: 'gradletestrunnersmoketest'
-      - name: "Run Integration test via Gradle"
-        uses: broadinstitute/datarepo-actions/actions/main@0.62.0
+      - name: "Run integration tests via Gradle"
+        uses: broadinstitute/datarepo-actions/actions/main@dr-2757-nm-better-unit
         with:
           actions_subcommand: 'gradleinttest'
           pgport: ${{ job.services.postgres.ports[5432] }}
           test_to_run: 'testIntegration'
+      - name: "Clean state lock from used Namespace on API deploy"
+        if: always()
+        uses: broadinstitute/datarepo-actions/actions/main@dr-2757-nm-better-unit
+        with:
+          actions_subcommand: 'k8_checknamespace_clean'
+      - name: "Clean whitelisted Runner IP"
+        if: always()
+        uses: broadinstitute/datarepo-actions/actions/main@dr-2757-nm-better-unit
+        with:
+          actions_subcommand: 'gcp_whitelist_clean'
+  publish_test_reports:
+    name: "Save execution reports and notify"
+    timeout-minutes: 60
+    needs:
+      - test_connected
+      - deploy_test_integration
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+    if: always()
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: "Load unit test cache"
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CACHE_PATHS }}
+          key: ${{ runner.os }}-build-unit
+      - name: "Load connected test cache"
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CACHE_PATHS }}
+          key: ${{ runner.os }}-build-connected
+      - name: "Load integration test cache"
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CACHE_PATHS }}
+          key: ${{ runner.os }}-build-integration
       - name: "Archive code coverage results"
```

Comment:
Here's how we did something similar for data catalog: https://github.com/DataBiosphere/terra-data-catalog/blob/main/.github/workflows/nightly-tests.yml#L121